tools/generate_report.rb

```diff
@@ -1,31 +1,52 @@
 #!/usr/bin/env ruby
 require File.expand_path('../config/environment', __dir__)
+require 'optimist'
 
 include ActionDispatch::Routing::UrlFor
 include Rails.application.routes.url_helpers
 
-USER_ID       = "admin".freeze
-REPORT_PARAMS = {:userid => USER_ID, :mode => "async", :report_source => "Requested by user"}.freeze
+options = Optimist.options(ARGV) do
+  banner "USAGE:  #{__FILE__} -u <name of userid which runs report, default: admin>              \n" \
+         "        #{__FILE__} -n <name of report or id of report, default: last created report>  \n" \
+         "        #{__FILE__} -l <log level, if this option is set log messages are forwarded to \n" \
+         "                                       standard output, allowed levels are debug, info, warn, error>  \n" \
 
-def report_from_args
-  if ARGV.empty?
+  opt :user, "userid", :short => "u", :type => String, :default => "admin"
+  opt :report_name_or_id, "report name or report id", :short => "n", :type => :string
+  opt :log_level, "log level (debug, info, warn, error)", :short => "l", :type => :string
+end
+
+# backward compatibility
+if ARGV[0].present? && options[:report_name_or_id].nil?
+  options[:report_name_or_id] = ARGV[0]
+end
+
+if options[:log_level]
+  Optimist.die "Log level #{options[:log_level]} is not supported, supported levels are: debug, info, warn, error" unless Vmdb::LogProxy::LEVELS.include?(options[:log_level].to_sym)
+  $log = Logger.new(STDOUT)
```

Comment:
Should this be a VmdbLogger or similar?

Reply:
yes, VMDBLogger (I saw in other scripts in tools)